### ISSUE TITLE
Explore/Spike: Blockchain Threshold Scheme 

### ISSUE THREAD

**Issue by fishseabowl** [pyrsia/pyrsia#400]
At the moment, threshold scheme(threshold signature scheme, threshold secret scheme) is the key component of ABFT consensus algorithm. We need to figure out the details and choose one solution.

Some Candidates:

Shamir's Secret Sharing, Ref: https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing
Application: Coinbase Custody (https://nomics.com/blog/flippening/coinbase-custody-sam-mcingvale)

BLS digital signature, Ref: https://en.wikipedia.org/wiki/BLS_digital_signature
Applications: Aleph BFT(https://arxiv.org/pdf/1908.05156.pdf), Honey Badger BFT(https://github.com/poanetwork/hbbft)

Schnorr signatures, Ref: https://en.wikipedia.org/wiki/Schnorr_signature
Application: Bitcoin(https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki)

Zero knowledge proofs, under research

**Comment by prince-chrismc** [comment]
If you are looking for a highlevel overview, i'd suggest https://academy.binance.com/en/articles/threshold-signatures-explained.

TL;DR

Public and Private keys are the norm today with asymmetric cryptography but there are other options which are better suited to blockchain's "zero-trust" mindset or philosophy 

***

Having to record all the public keys is cumbersome. Our first draft solution is to store them on the blockchain

### TSS

 allows "zero-proof" trust... in place of trusting the public keys that are recorded

> a set of parties that do not trust each other try to jointly compute a function over their inputs while keeping those inputs private. 

### SSSS

> provides a way to store the private key in a distributed manner such that while the private key is at rest, it is stored in multiple locations

**Comment by chb0github** [comment]
We've discussed and reviewed and we are going to do as Aleph does. We will document and share 

[BLS digital signature](https://en.wikipedia.org/wiki/BLS_digital_signature)
Applications: [Aleph BFT](https://arxiv.org/pdf/1908.05156.pdf), [Honey Badger BFT](https://github.com/poanetwork/hbbft)

**Comment by prince-chrismc** [comment]
#414
